### DISCUSSION TITLE
Scenario does not continues after a loop

### DISCUSSION THREAD

**Discussion by matiaszc** [artilleryio/artillery#1175] [Artillery]
Hello,

I have a yaml script that make a login post in the before section and then in scenario section upload two files to my server. I have to check if the first file was upload correctly before upload the second one. I had tried to ways:

1. Without nested loops:
    - upload of the first file
    - check the status within a loop
    - upload the second file
    - check the status within a loop

2. Wiht nested loops:
    - start loop
    - upload of the first file
    - check the status within a loop
    - count: 2

In both cases scenario ends after the first loop, that results in only one file uploaded.

Have anyone experience something similar?

Thanks in advance.

**Comment by hassy** [comment]
hi @matiaszc 👋 can you share the script, or at least part of it?

**Reply by matiaszc** [reply]
Hi @hassy 

thank you for your reply. Here are part of the script

I make the login in a before section

    #Upload first file
    - post:
        url: "/submit"
        headers:
          Content-Type: "application/json"
        json:
          "userid": "{{ uid }}"
          "userkey": "{{ ukey }}"
          "audiodata": 
        capture:
          - json: "$.jobid"
            as: "jobid"
          - json: "$.jobkey"
            as: "jobkey"
          - json: "$.jobwait"
            as: "jobwait"
        
    - loop:
        - think: "{{ jobwait }}"
        - post:
            url: "/results"
            headers:
              Content-Type: "application/json"
            json:
              "jobid": "{{ jobid }}"
              "jobkey": "{{ jobkey }}"
            capture:
              - json: "$.result"
                as: "result"
              - json: "$.jobwait"
                as: "jobwait"
      whileTrue: "statusReady"
      
    #Upload second file
    - post:
        url: "/submit"
        headers:
          Content-Type: "application/json"
        json:
          "userid": "{{ uid }}"
          "userkey": "{{ ukey }}"
          "audiodata": 
        capture:
          - json: "$.jobid"
            as: "jobid"
          - json: "$.jobkey"
            as: "jobkey"
          - json: "$.jobwait"
            as: "jobwait"

    - loop:
        - think: "{{ jobwait }}"
        - post:
            url: "/results"
            headers:
              Content-Type: "application/json"
            json:
              "jobid": "{{ jobid }}"
              "jobkey": "{{ jobkey }}"
            capture:
              - json: "$.result"
                as: "result"
              - json: "$.jobwait"
                as: "jobwait"
      whileTrue: `"statusReady"

I verify the status of result field with this js function (that part work well)

functions.js                                                                
module.exports = {
  statusReady: statusReady
}

function statusReady(context, next) {
  const continueLooping = context.vars.jobstatus !== 'success';
  // While `continueLooping` is true, the `next` function will
  // continue the loop in the test scenario.
  return next(continueLooping);
}

The problem is the scenario only upload one file. I tried looping with count: 5 with the same result, after the first loop ends the scenario ends and continues with the next user. I also tried the loop without end condition (no count, no whileTrue) an the loop finish anyway. The response of file upload operation contains a field that returns 'complete' as status. I think that this field has something to do because the loop does not ends until I got this response. I don't capture this field.

Thank you in advance

**Reply by hassy** [reply]
Is it possible that the response from `/submit` does not always include the fields which are captured with `capture`? Artillery will stop the virtual user if a capture fails by default, and that might be what's happening.

**Reply by matiaszc** [reply]
Hi @hassy 

The response always return the field result from /submit, that field can contains several status (failure, pending, complete) but is always returned. The only constant that I could detects is that the loop always ends when the field return the string 'complete', even if there is no count or whileTrue condition.

Thanks again for your reply
